test/apps/iodemo/io_demo.cc

```diff
@@ -1660,10 +1663,20 @@ class DemoClient : public P2pDemoCommon {
         }
 
         if (!disconnecting) {
+            // remove active servers entry
+            if (server_info.active_index !=
+                std::numeric_limits<size_t>::max()) {
+                active_servers_remove(server_index);
+            }
+
             /* Destroying the connection will complete its outstanding
              * operations */
             server_info.conn->disconnect(new DisconnectCallback(*this,
                                                                 server_index));
+        } else {
+            // server must be removed from the list of active servers
+            assert(server_info.active_index ==
+                   std::numeric_limits<size_t>::max());
```

Comment:
minor: this assertion can be outside of 'else'. e.g:
```
if (!disconnecting) {
  active_servers_remove
  disconnect
}

assert(active_index == max() )
```

Reply:
done